kong/dao/schemas/apis.lua

```diff
@@ -83,8 +83,13 @@ local function check_uri(uri)
     return false, "invalid"
 
   elseif not match(uri, "^/[%w%.%-%_~%/%%]*$") then
-    -- Check if characters are in RFC 3986 unreserved list, and % for percent encoding
-    return false, "must only contain alphanumeric and '., -, _, ~, /, %' characters"
+    -- URI contains characters outside of the reserved list of
+    -- RFC 3986: the value will be interpreted as a regex;
+    -- but is it a valid one?
+    local _, _, err = ngx.re.find("", uri)
```

Comment:
Right now it is probably not going to matter, but there are flags, that I think may change how regexes are interpreted (e.g. `x`). I was just thinking it would be good if we follow the same arguments here. Mostly because there is no harm adding them.